sql/analyzer/validation_rules.go

```diff
@@ -46,6 +53,7 @@ var DefaultValidationRules = []Rule{
 	{validateSchemaSourceRule, validateSchemaSource},
 	{validateProjectTuplesRule, validateProjectTuples},
 	{validateIndexCreationRule, validateIndexCreation},
+	{validateCaseResultTypesRule, validateCaseResultTypes},
```

Comment:
Just open question - doesn't make sense as a optimization to squash some rules?
For instance instead of 3 times looping, we may have one loop with some cases - loud thinking.

Reply:
this query phase is really fast compared with the rest. The idea of having separated rules is the synergy that we can obtain executing them again and again, interacting with other rules.